lwt_domain.opam

```diff
@@ -12,6 +12,8 @@ depends: [
   "base-domains"
   "domainslib" {>= "0.5"}
   "lwt" {>= "5.6"}
+  "ocaml-lsp-server" {with-dev-setup}
```

Comment:
I don't think that belongs here. Devs might decide to use merlin instead, or just no such features.

It's not like ocamlformat which has some build rules that depend on it, and a CI that'll complain if it wasn't used.

Reply:
Sure